src/os_win/os_fs.c

```diff
@@ -86,23 +86,11 @@ __win_fs_rename(WT_FILE_SYSTEM *file_system,
 	WT_ERR(__wt_to_utf16_string(session, from, &from_wide));
 	WT_ERR(__wt_to_utf16_string(session, to, &to_wide));
 
-	/*
-	 * Check if file exists since Windows does not override the file if
-	 * it exists.
-	 */
-	if (GetFileAttributesW(to_wide->data) != INVALID_FILE_ATTRIBUTES)
-		if (DeleteFileW(to_wide->data) == FALSE) {
-			windows_error = __wt_getlasterror();
-			__wt_errx(session,
-			    "%s: file-rename: DeleteFileW: %s",
-			    to, __wt_formatmessage(session, windows_error));
-			WT_ERR(__wt_map_windows_error(windows_error));
-		}
-
-	if (MoveFileW(from_wide->data, to_wide->data) == FALSE) {
+	if (MoveFileExW(from_wide->data,
```

Comment:
There is a slightly offputting article here about the semantic of MoveFileExW with MOVEFILE_REPLACE_EXISTING:
http://www.tek-tips.com/viewthread.cfm?qid=570894

I can't find the knowledge base article it refers to, and it's pretty old, so maybe it's not relevant?

Possibly more relevant:
http://stackoverflow.com/a/167475

Which says:
```
Setting this value guarantees that a move performed as a copy and delete operation
is flushed to disk before the function returns. The flush occurs at the end of the
copy operation.
```

That's not reassuring, because what happens if there is a hard power failure between the copy and delete?